android/src/main/cpp/graphics/shader/BaseShaderProgramOpenGl.cpp

```diff
@@ -68,18 +70,58 @@ void BaseShaderProgramOpenGl::checkGlProgramLinking(GLuint program) {
     }
 }
 
+const GLuint BaseShaderProgramOpenGl::FRAME_UBO_BINDING_POINT = 0;
+
+/*
+ * Returns the uniform block definition for the frame-wide render values.
+ * Caution! The binding location is fixed to BaseShaderProgramOpenGl::. To use this buffer and to avoid collisions with other UBOs, other shaders
+ * must comply.
+ *
+ * (Padded to 16 bytes due to std140!)
+ */
+const std::string BaseShaderProgramOpenGl::FRAME_UBO_DEFINITION =
+        "layout(std140, binding = " + std::to_string(FRAME_UBO_BINDING_POINT) + ") " + OMMShaderCode(uniform FrameUniforms
+        {
+            mat4 vpMatrix;
+            vec4 origin;
+            vec2 frameSpecs; // x: screenPixelAsRealMeterFactor, y: timeFrameDeltaSeconds
+        } uFrameUniforms;
+);
+
+void BaseShaderProgramOpenGl::setupFrameUniforms(GLuint frameUniformsBuffer, int64_t vpMatrix, const Vec3D &origin,
+                                                 double screenPixelAsRealMeterFactor, double timeFrameDeltaSeconds) {
+    glBindBuffer(GL_UNIFORM_BUFFER, frameUniformsBuffer);
+    glBufferSubData(GL_UNIFORM_BUFFER, 0, 4 * 4 * sizeof(GLfloat), (GLfloat *)vpMatrix);
+    tempVec4FrameUniforms[0] = origin.x;
```

Comment:
Nit: would seem nicer to avoid this tempVec4FrameUniforms and allocate it on the stack instead, e.g.
```
std::array<GLfloat, 4> temp;
```
Btw. wouldn't it make sense combine the BufferSubData calls (by making the temp array 6 floats wide).